### ISSUE TITLE
Incompatible with 1.6.640

### ISSUE THREAD

**Issue by whoot** [ArranzCNL/ImprovedCameraSE#42]
Could you please update the mod so it works with the new Skyrim version 1.6.640?

Thanks in advance!

**Comment by ArranzCNL** [comment]
It will be updated soon, it is just working out what to do with the 353 branch.

**Comment by SpiritOfSuffering** [comment]
> Could you please update the mod so it works with the new Skyrim version 1.6.640?

На самом деле, вы не должны использовать эту версию Skyrim, т.к. она просто будет несовместима с большинством модов, которые уже существуют. Используйте 1.6.353, и не выше.

**Comment by whoot** [comment]
> > Could you please update the mod so it works with the new Skyrim version 1.6.640?
> 
> На самом деле, вы не должны использовать эту версию Skyrim, т.к. она просто будет несовместима с большинством модов, которые уже существуют. Используйте 1.6.353, и не выше.

I know that, but hopefully most mods will be updated in the coming couple of weeks so it should work again.

**Comment by Veradra** [comment]
> It will be updated soon, it is just working out what to do with the 353 branch.

It may be worth looking into CommonLibSSE(-NG). Once upstream has updated to the newer game version, updating the mod _should_ be relatively simple if/when a new version is released.

Obviously, it requires having another dependancy and possibly a decent amount of work to move over to it, but it could be a way to avoid worrying about inter-version compatibility.

**Comment by barcharcraz** [comment]
The 640 update breakage is only sorta breakage, address library updated quickly, and however Arranz gets address mappings should also work. But some data structures in the game were changed, so plugins must now say that they either never depended on the layout of those structures or that they have updated their definitions

**Comment by ArranzCNL** [comment]
@Veradra
I've looked into CommonLibSSE-NG it downloaded over 70k+ worth of files which was around 5gig of data, I mean what!?. So no thankyou rather stick to Visual Studio unless I am making a multi-platform project then I would consider cmake. Feels like you need a degree to understand how cmake works.

@barcharcraz 
The addresses were easy to fix and to be honest only a few needed updating. Just need to alter the way in which the universal side works reason for waiting and not being hasty is incase SKSE decided to do another mod breaking update.

**Comment by barcharcraz** [comment]
Odd, my copy of CommonLib-NG is only 26MB total (including the git history). You need not convert your project to cmake to use CommonLib-NG btw, and actually, if you use it with vcpkg you don't really need to touch cmake at all, vcpkg integrates directly with msbuild/visual studio projects. Still, if you already have a cross-version thing going on I it might be easiest to just continue to do that!

**Comment by ArranzCNL** [comment]
I had to switch the vcpkg location as it tried blowing up my SSD when messing with NG, so monitoring what was being installed yeah as I mentioned previously.

The way I have it designed for Visual Studio 2019/2022 click on your relevant solution file and just build no fraffing around etc even your granny can build it without having to mess around with vcpkg and cmake! Only 1 prerequisite Windows 10 SDK 10.0.20348.0, as there is some form of bug with previous versions which prevents MinHook from building.

**Comment by netizen-a** [comment]
will there be an update for the GOG edition as well?

**Comment by ArranzCNL** [comment]
@Razordor
Time of writing this no as would need access to the binary to analyse and no doubt possible structure changes again with SKSE.

**Comment by d1zzy** [comment]
@ArranzCNL Do you need only the executable or the full game?

Do you have an email address or some other form of private communication? Thanks!

At this time this this mod pretty much the only major functionality I'm lacking for my GOG Skyrim AE copy :)

**Comment by ArranzCNL** [comment]
Well thanks to Noughtmare's tool we have the addresses for GOG now for Improved Camera at least.

The main reason for the delay of releasing is knowing what to do about SKSE or switch to CommonLibSSE. Due to SKSE doing architectural updates recently.

**Comment by whoot** [comment]
so, you can start updating it? Sounds great!

**Comment by LifeWulf** [comment]
Eagerly anticipating the next update! So jarring to look down and just be floating.

**Comment by Nuanulla** [comment]
So there’s certainly no obligation for anyone to respond or even continue on this mod. And you’d be in the right to do so or remove this comment. But that cannot discount the fact that there’s a shocking lack of transparency here. Still hoping for the best.

**Comment by cetedus** [comment]
I will say what the author perhaps does not want/can't say - feel free to fork the project and continue development. The license permits it. 
Otherwise - wait. Tt will be updated or it will not be. Or you can downgrade Skyrim to ver. 353 and use the currently available version of the mod.

Sorry for butting in, but I can see it here as well as in other places (Nexus in particular) that people think they are somehow entitled to get updates simply because they want them.

**Comment by Veradra** [comment]
> feel free to fork the project and continue development.  The license permits it.

Non-starter, the mod is closed source. What the license says does nothing if it requires a full decomp anyways.

**Comment by Nuanulla** [comment]
> Sorry for butting in, but I can see it here as well as in other places (Nexus in particular) that people think they are somehow entitled to get updates simply because they want them.

The earlier paragraph is the response I'd rather hear from the author. 

I also see those comments everywhere for all kinds of mods—that's a conversation I refuse to join. I'm not expecting an update per se, but am keeping track of it, so I've attempted to separate myself from those other commenters. Transparency is often a scarce resource and not for the right reasons. If progress is halted, I'd rather hear it directly that I should no longer expect any news rather than come to that conclusion myself. That's my right to believe, but not something I should expect to be entitled to. 

Basically, I recognize transparency is never itself a strict right, and only a privilege. But I think everyone else would be grateful to hear something rather than nothing.

**Comment by cetedus** [comment]
> > feel free to fork the project and continue development.  The license permits it.
> 
> Non-starter, the mod is closed source. What the license says does nothing if it requires a full decomp anyways.

Ah you are right, got misled by the number of forks already in existence.

**Comment by ArranzCNL** [comment]
Source isn't available yet, it will be once released on Nexus under MPL 2.0 license.

I do apologise for lack of updates/news jist of it is the interface changed on SKSE which still needs resolving. Purposely waited to do anything for a couple of weeks due to anymore changes which there were as GOG version came out. Originally ImprovedCameraSE was two seperate repo's but managed to sort the projects out so they could co-exist without hassle under one repo. Then I just lost interest for a few weeks.

Currently testing and fixing the remaining com issues to do with the menu before moving on to getting 1.6.640 and 1.6.659 working. The other backend changes I need to do can wait for the release after.

**Comment by Xerxes2222** [comment]
@ ArranzCNL, is there any way to support you besides saying Thank You! I just updated to 1.6.640 and I didn't realize how important improved camera is to me until I was without it....please keep up the good work, looking forward to using improved camera again

**Comment by srwaggon** [comment]
Sorry for the spam, but I came by to say that this mod is easily one of the most amazing mods for Skyrim. I've recently updated my AE install to 1.6.640 and I sorely miss this mod. I'm eagerly looking forward to an updated release and I'd love to contribute in any way.

**Comment by ArranzCNL** [comment]
If you can test hotfix 3 that will be of great benifit. Hoping to have this out in the next few days since the worst part is over with! Contains significant changes for the migration and just plugin design in general imho.

**Comment by VartStolpahus** [comment]
when will hotfix 3 be released?

**Comment by Bulljump** [comment]
I have Skyrim AE 1.6.640, but am running SKSE at 1.5.97. Improved Camera 5 works just fine for me.

**Comment by TheMotonaut** [comment]
> If you can test hotfix 3 that will be of great benifit. Hoping to have this out in the next few days since the worst part is over with! Contains significant changes for the migration and just plugin design in general imho.

Do you need a tester? I am up for testing. I am running 640 with papurys, SKSE and etc.

**Comment by cppcooper** [comment]
Aye, I think you've got a boat load of testers ready and willing! Just post it somewhere (in releases?)

edit: just figured out what's going on, ie. prep work for 640 (which needs testing)

**Comment by grillzman** [comment]
Will there be an update for 1.6.640? I thought this most recent update was going to be for 1.6.640 .

**Comment by abishai** [comment]
@Bulljump That's not possible - 1.5.97 won't work with 1.6.640

**Comment by ArranzCNL** [comment]
@grillzman work is on that now, hotfix 3 was the needed backend changes required for making porting as painless as possible.

**Comment by grillzman** [comment]
No rush. Just making sure I wasn't a moron. Thanks for the reply.

**Comment by LoneWolf13778** [comment]
So I saw that reshade is a requirement if I want the customization menu. Is there any way to still use an enb and get just the reshade menu?

**Comment by cppcooper** [comment]
@LoneWolf13778 not what this github issue is for, please don't ping everyone on the thread to ask about enbs

**Comment by LoneWolf13778** [comment]
@cppcooper sorry never really used github before and wasn't sure where to ask or what to do and also wanted to say I wanted to test for the AE update

**Comment by ArranzCNL** [comment]
Just to give a brief update on progress:

- [x] Address ID's 1.5.97
- [x] Address ID's 1.6.353
- [x] Address ID's 1.6.640
- [x] Address ID's 1.6.659
- [x] Double check ID's
- [x] CommonLibSSE-NG library (Will be sledgehammered in initially!)
- [x] PR5 project files fixed. Binaries won't be released, if anyone does compile when source is available they are half the size.
- [x] Minhook minor fixes.
- [x] mINI minor fixes.
- [ ] Convert code over.
- [x] Drop support for VC16 (Visual Studio 2019)
- [x] C++ 20. PR5 now supports this but won't be released.
- ~~Look into crashing issues with menu, under certain conditions. (More important for GOG 1.6.659)~~

I'll tick things once done (thanks @noughtmare for pointing that out!), hoping to have all the address id's by the end of tonight.

**Comment by noughtmare** [comment]
@ArranzCNL You can create tickboxes like this:

```
- [x] Address ID's 1.5.97
- [ ] Address ID's 1.6.353
```

- [x] Address ID's 1.5.97
- [ ] Address ID's 1.6.353

**Comment by Markharvey9889** [comment]
Hi,
really impressed with your mod :) quick question - when will the new version release for 1.6.640?
Cheers,
Mark

**Comment by cppcooper** [comment]
Updates and checklists are no longer sufficient! GIVE US A LIVESTREAM OF YOUR LIFE!
/sarcasm

**Comment by the-r3dacted** [comment]
What's the point of that reply?

**Comment by jimmon89** [comment]
because of people not taking the time to read the thread and notice the checklist.
patience is a virtue
to read the thread is to be one with the gods

**Comment by the-r3dacted** [comment]
Oh yeah I see that. I've just been following all activity by email waiting for release.

**Comment by jimmon89** [comment]
same, but you weren't the one asking when the mod will be done

**Comment by cppcooper** [comment]
I mean, there are only two definite articles left on that list. (The bug sounds like it's probably difficult to reproduce)

For whatever it is worth:
I would hazard a guess that integrating commonLibSSE-NG will take the most time (few hours?), as for C++20 I'm very curious what features are even desired.. But alas no source.

**Comment by jimmon89** [comment]
most mod authors will only tell people that it will release "Soon™" or "When it's done™" and give zero information on what is going on behind the scenes.
the especially petty ones will say that every time someone asks about the mod releasing, the release will be delayed by a day(some say week or even month)
we should count ourselves lucky the we even got a checklist, and patiently wait for the author to let us know when it's done.

**Comment by LeBlanc776** [comment]
I just want to know how much time is left until it releases. I don't want to sound impatient but I can't really play without these type of mods and unfortunately, Joy of Perspective is super buggy and IFPV doesn't even have support anymore if I recall correctly, doesn't work properly either anyways. This one is pretty much the only option left 💀

**Comment by ArranzCNL** [comment]
As I updated the checklist to include Sledgehammer wording, that is what I shall be doing initially with CommonLibSSE-NG. Imagine CLibNG is a square and IC only has a circle slot, yeah... however you can thank [gottyduke](https://github.com/gottyduke) to get the parts I require. Address ID's would have taken the longest to acquire and this stage now to get the library inside and building.

Once the library is inside and building correctly, should be a 1-2 days to switch the codebase over. It won't be a full release on nexus yet as I will want to redo the CLibNG library and shall release that separately over here as well. (Flatrim with no cmake useless fluff)

Edit:
As for the crashing issues with the menu with current builds, I was unable to reproduced (wasted a day) trying out varying combinations and my only conclusion to this is Rivatuner (not setup using MS Detours)/Something similar or people using a pirated copy of the game.

**Comment by SpikerzZ406** [comment]
Is it almost ready for 1.6.640?

**Comment by srwaggon** [comment]
> Is it almost ready for 1.6.640?

With love to all, I am coming here to ask everyone to stop posting to heckle the author regarding updates and finishing their gift to the community and to instead prefer silent appreciation and Patreon donations.

GitHub and especially issues is not the place for this type of discourse. These sorts of comments may bury relevant issue discussion and I urge folks to keep the space clean so the author may work and focus at their own pace.

Unfortunately, I'm guilty too. I may be entirely out of place, but I appreciate what this author has given to the community. Let's respect their gift and return it with patience, donations, and workspace cleanliness. I urge using other platforms such as Reddit or Twitter for these sorts of questions and comments.

**Comment by SpikerzZ406** [comment]
> > Is it almost ready for 1.6.640?
> 
> With love to all, I am coming here to ask everyone to stop posting to heckle the author regarding updates and finishing their gift to the community and to instead prefer silent appreciation and Patreon donations.
> 
> GitHub and especially issues is not the place for this type of discourse. These sorts of comments may bury relevant issue discussion and I urge folks to keep the space clean so the author may work and focus at their own pace.
> 
> Unfortunately, I'm guilty too. I may be entirely out of place, but I appreciate what this author has given to the community. Let's respect their gift and return it with patience, donations, and workspace cleanliness. I urge using other platforms such as Reddit or Twitter for these sorts of questions and comments.

I just asked a question, of course I appreciate the author, I would donate to him no problem if I had extra money too, he is one of the few that I would, people are aloud to be curious, he doesn't owe us anything, I know this as a game designer and Modder this whole topic is about it not working with 1.6.640 so I thought I would ask, it is aloud.

have a good day :D

**Comment by LeBlanc776** [comment]
> > Is it almost ready for 1.6.640?
> 
> With love to all, I am coming here to ask everyone to stop posting to heckle the author regarding updates and finishing their gift to the community and to instead prefer silent appreciation and Patreon donations.
> 
> GitHub and especially issues is not the place for this type of discourse. These sorts of comments may bury relevant issue discussion and I urge folks to keep the space clean so the author may work and focus at their own pace.
> 
> Unfortunately, I'm guilty too. I may be entirely out of place, but I appreciate what this author has given to the community. Let's respect their gift and return it with patience, donations, and workspace cleanliness. I urge using other platforms such as Reddit or Twitter for these sorts of questions and comments.

While I agree with you wholeheartedly I can also see why people would get impatient. As I stated before some time ago, this is the only first person mod that could work for most people, as the other options (JoP and IFPV) are well... Not very functional.

I just hope we, users of 1.6.640 get a working release soon. Specially us with OCD that can't play with these types of mods

**Comment by ArranzCNL** [comment]
Just to give an brief update I am having major issues linking to the library at present, don't know when it'll be fixed as I only have finite free time.

**Comment by eli-j-s** [comment]
i figured it was finally time to update to 1.6.6 and start fresh.  i was wrong as this mod is essential for first person players :(

**Comment by diegito55** [comment]
awe man im late to the party, but still no update for the 1.6.640??

**Comment by somewhereovertherain** [comment]
> awe man im late to the party, but still no update for the 1.6.640??

yet.

**Comment by the-r3dacted** [comment]
What is C++ 20 for?

**Comment by killzoms** [comment]
> What is C++ 20 for?

Game updated with the Anniversary update, this changed the C++ standard that was used by the game

**Comment by cppcooper** [comment]
Ahh, so it is just about being ABI compatible, not using any new features. Thanks for that info =)

**Comment by Kurnsh** [comment]
If I may ask, if we play without improvedcamera dll now on 1.6.640., will it be possible to add it mid game later on?

**Comment by Veradra** [comment]
> If I may ask, if we play without improvedcamera dll now on 1.6.640., will it be possible to add it mid game later on?

In general, you can add/remove .DLL mods mid playthrough.

**Comment by parsafaramarzi** [comment]
eagerly waiting for this

**Comment by GoldMonkey8** [comment]
checking it everyday ':)

**Comment by the-r3dacted** [comment]
![ScreenShot_20230406022114](https://user-images.githubusercontent.com/77039240/230303981-7842888b-bd46-4e38-b5e4-4a173542f816.png)

**Comment by Abab-bk** [comment]
"I am so excited for version 1.6.460, can't wait! 😍😍😍😍

**Comment by Soldream0** [comment]
I just want to say to the author that you should not overload yourself because of the impatience of people. If you stick to your pace, it will be better for everyone. The main thing is that the mod turned out to be stable, and the process does not destroy your psyche. People will wait as long as it takes.

**Comment by GoldMonkey8** [comment]
yep, please take your time. I'm just glad its going to come out.

**Comment by ArranzCNL** [comment]
Just to give a brief update on what is going on since I have been silent over the past week or so.

Project file is broken (VS project) don't know why. This is easily fixable but it will mean the current released build will be locked to VS 2022 for compilation as I have no intention to get VS 2019 working with it once fixed. (All released binaries are built using VS 2022 anyhow). Mentioning this as source code will be released so it is off the dev branch which can focus solely on NG, this will happen when NG release occurs.

As for the initial NG release it will be sledgehammered in as previously mentioned. Some issues/errors need fixing with NG and also with mINI library (trivial). Fixed initial linking error with NG which was kind of my bad not reading the header files but then again why not name the interface for plugins properly! Source code to this version will be released once I get CommonLib converted to VS 2022 without cmake and vcpkg which is completely pointless imho adding an extra layer of unnecessary complexity to modding a single game just for Windows.

I cannot give a reasonable eta currently as I have a family member in hospital hence my absence.

**Comment by Baggytrousers27** [comment]
Completely understandable. Please send them our love and stay safe yourself.

**Comment by cppcooper** [comment]
Clion is a thing, and relies soley on cmake. So I wouldn't say completely pointless, given how well clion works. Vcpkg though, that seems extra to me.. Like whats wrong with extern/ submodules?? Works great

**Comment by siaccarino** [comment]
Somebody else did already a port to VS 2022:
https://github.com/SkyrimScripting/SKSE_Templates

I can work very well with this port, it is pretty easy to use.

**Comment by siaccarino** [comment]
> once I get CommonLib converted to VS 2022

Somebody else did already a port to VS 2022:
https://github.com/SkyrimScripting/SKSE_Templates

I can work very well with this port, it is pretty easy to use.

**Comment by Zphre** [comment]
@ArranzCNL ^^^

**Comment by ArranzCNL** [comment]
@siaccarino That isn't a port to VS 2022 but it doesn't not matter as what I currently have works, once the next release is out I'll fix up what I have done and release the source files. It will be up to Po3 and/or CharmedBaryon if they want to include although I'll have the rendition over here.

The purpose in what I am/will be doing with this is to make it as beginner friendly as possibly, I know @cppcooper mentioned Clion which is fine but they are obviously more competent than a beginner. My philosophy is to KIS (keep it simple), average user should just be able to install Visual Studio 2022 and compile code not screw around with other build systems, there is a time and place to use different build systems but I'll reiterate on what I mentioned previously, 1 single game which is Windows only is not one of them!

**Comment by ArranzCNL** [comment]
Just a brief update the sledgehammer approach is completed at least until I run into any other issues.

Implemented a logging system via spdlog (thanks to CLib) which is quite nifty, I won't lie I was cussing at it for a few hours as the syntax is rather weird. End user will only get to experience the file logging side of it. (you won't see a difference)

Next on the agenda is to get the hooks back in place and do up some form of dumper to check addresses are pointing to where they should for various skyrim versions. Once this is completed it is just a case of switching the remaining code to CLib.

Screenshot is from 1.6.640... so soon, can't say how soon other than soon!
![ic_ng](https://user-images.githubusercontent.com/52199624/232923755-1c6959ab-2439-459e-b9c4-f0205d267678.png)

**Comment by LifeWulf** [comment]
Glad to hear we’re almost there! Thank you for all your hard work!On Apr 18, 2023, at 19:22, ArranzCNL ***@***.***> wrote:﻿
Just a brief update the sledgehammer approach is completed at least until I run into any other issues.
Implemented a logging system via spdlog (thanks to CLib) which is quite nifty, I won't lie I was cussing at it for a few hours as the syntax is rather weird. End user will only get to experience the file logging side of it. (you won't see a difference)
Next on the agenda is to get the hooks back in place and do up some form of dumper to check addresses are pointing to where they should for various skyrim versions. Once this is completed it is just a case of switching the remaining code to CLib.
Screenshot is from 1.6.640... so soon, can't say how soon other than soon!

**Comment by LoneWolf13778** [comment]
@ArranzCNL Glad to hear that it's almost ready, thank you for all the work you've put into it. I was wondering though, will the update be put into the ImprovedcameraSE or AE file?

**Comment by Dreikantbrot** [comment]
> @ArranzCNL Glad to hear that it's almost ready, thank you for all the work you've put into it. I was wondering though, will the update be put into the ImprovedcameraSE or AE file?

I'm assuming you mean the ImprovedCameraSE and ImprovedCameraAE repositories? In that case, it'll most likely be here, since the [AE repository has been deprecated](https://github.com/ArranzCNL/ImprovedCameraAE/releases) in favor of the SE repo. Afaik, everything will be released here for both SE and AE.

**Comment by ArranzCNL** [comment]
@LoneWolf13778 That is a good point I should remove the AE repo! I was thinking of creating NG but decided against it so it will be hosted under SE (here) which makes sense.

I'll do my best to try and make it less confusing as I believe Preview Release 5 final will be under releases and NG under tags. The reason for this is due to source code being released for PR5 which will be on the main branch until I am ready to release NG's source code in which case that will take over main branch, it'll be released on Nexus anyway then as v1.0.1.7871.

For those not too savvy on github when v1.0.1.7871 is released, this will be under main branch, preview release 5 will have it's own branch and there will be a dev branch for upcoming updates to main branch - so bleeding edge additions/crashes/fixes. Just to note I will accept pull requests where it makes sense for PR5 as the architecture is similar until 1.0.1.7871 source is released.

I'll update again when I know the hooks are working 100% then it should be 1-3 days for release.

**Comment by LoneWolf13778** [comment]
@ArranzCNL thank you for letting me know. I just wasn't sure if the new update and coding would work on SE like AE so was just curious on which repo it would be in. Again thank you for your response and all the work you have put into this mod I know a lot of us appreciate it.

**Comment by ShaheerVD** [comment]
Any update on the progress for updating it to 640?

**Comment by Nuanulla** [comment]
Everyone: kindly stop replying to this GitHub open issue with shallow comments. For those subscribed, the messages amount to spam. For those unsubscribed, they return to productive conversations nested within unnecessary fluff. This is not a forum thread. 

But, I guess any others concerned might as well “watch” only for releases like @K4sum1 has done.

**Comment by cppcooper** [comment]
> But, I guess any others concerned might as well “watch” only for releases like @K4sum1 has done.

There's also this option, but beware after making new comments it reverts to a regular subscription.
![image](https://user-images.githubusercontent.com/7276349/234642338-dc2ab203-aaf2-4341-8c3f-371a4d983694.png)

**Comment by Baggytrousers27** [comment]
> Everyone: kindly stop replying to this GitHub open issue with shallow comments. For those subscribed, the messages amount to spam. For those unsubscribed, they return to productive conversations nested within unnecessary fluff. This is not a forum thread.
> 
> But, I guess any others concerned might as well “watch” only for releases like @K4sum1 has done.

Surely you recognise the irony in posting this 3 days after the last comment?

> > But, I guess any others concerned might as well “watch” only for releases like @K4sum1 has done.
> 
> There's also this option, but beware after making new comments it reverts to a regular subscription. ![image](https://user-images.githubusercontent.com/7276349/234642338-dc2ab203-aaf2-4341-8c3f-371a4d983694.png)

Cheers for this, makes things much simpler.

**Comment by Nuanulla** [comment]
> Surely you recognise the irony in posting this 3 days after the last comment?

No, not at all. Don't think you're so smart with your rhetorical question. It'd be worse if you were subscribed like 40 of us yesterday because you can still see the reply posted 2 hours before my last one, still copied word-for-word in your inbox.

Kind reminder: Deleting your comments is always an option on Github. As is editing.

**Comment by SydByrd** [comment]
> Everyone: kindly stop replying to this GitHub open issue with shallow comments. For those subscribed, the messages amount to spam. For those unsubscribed, they return to productive conversations nested within unnecessary fluff. This is not a forum thread.
> 
> But, I guess any others concerned might as well “watch” only for releases like @K4sum1 has done.

![sdzxfvS (2)](https://user-images.githubusercontent.com/132075476/235165941-f9bbd9c3-d408-49ab-923e-362038dfc86a.png)

**Comment by Nightmare2121** [comment]
> 

Bruh i just want improved camera on 1.6.640 Lmao

**Comment by Nightmare2121** [comment]
new ETA?

**Comment by ArranzCNL** [comment]
Planned for 13th May, internally should be ready this weekend at some point.

**Comment by lupusvir** [comment]
I have a quick question. I was looking into the mod and found its .ini file. Inside it was a version. I tried changing the version but the mod keep insisting it was the other version. I was wondering if the skyrim version the mod can run on was baked into the mod and not just the .ini file.

**Comment by Veradra** [comment]
> I have a quick question. I was looking into the mod and found its .ini file. Inside it was a version. I tried changing the version but the mod keep insisting it was the other version. I was wondering if the skyrim version the mod can run on was baked into the mod and not just the .ini file.

The version in the ini is the minimum and maximum version that .dll supports. Don't change it, not because it'll necessarily do any harm, but because it's utterly pointless.

**Comment by lupusvir** [comment]
> > I have a quick question. I was looking into the mod and found its .ini file. Inside it was a version. I tried changing the version but the mod keep insisting it was the other version. I was wondering if the skyrim version the mod can run on was baked into the mod and not just the .ini file.
> 
> The version in the ini is the minimum and maximum version that .dll supports. Don't change it, not because it'll necessarily do any harm, but because it's utterly pointless.

Yeah, I figured that out. I was just hoping for a way to change the version so I could still use it. While waiting on the next version.

**Comment by K1ragami** [comment]
Is the update coming in today? Literally held up playing Skyrim today, cuz I've heard the mod update is coming.

**Comment by ACEROALMA** [comment]
> Is the update coming in today? Literally held up playing Skyrim today, cuz I've heard the mod update is coming.

Same here :)

**Comment by ArranzCNL** [comment]
Unlikely as I do want to check out the newest CLib-NG and I'm still not quite finished getting some hooks back in place. I do also want to correct some mistakes mainly naming ex. GetSitSleepState should be GetInteractiveState as the states are nothing to do with sitting etc. (They are but cover much more than just sitting!)

I might do a release on discord for testing purposes for the events as this is a very time consuming process.

**Comment by SkillzMan** [comment]
> Unlikely as I do want to check out the newest CLib-NG and I'm still not quite finished getting some hooks back in place. I do also want to correct some mistakes mainly naming ex. GetSitSleepState should be GetInteractiveState as the states are nothing to do with sitting etc. (They are but cover much more than just sitting!)
> 
> I might do a release on discord for testing purposes for the events as this is a very time consuming process.

Apologies to bother you, but what discord server would that be on if you do a test release? Could I possibly get a link?

**Comment by Crif3** [comment]
> Unlikely as I do want to check out the newest CLib-NG and I'm still not quite finished getting some hooks back in place. I do also want to correct some mistakes mainly naming ex. GetSitSleepState should be GetInteractiveState as the states are nothing to do with sitting etc. (They are but cover much more than just sitting!)
> 
> I might do a release on discord for testing purposes for the events as this is a very time consuming process.

Is your discord server private? would like to join and test too!

**Comment by LeBlanc776** [comment]
I would like to join the Discord server too 👀

**Comment by AndiBruehl** [comment]
Me too.

**Comment by Seath1337** [comment]
me too

**Comment by ArranzCNL** [comment]
[Discord Link](https://discord.com/invite/TqDzcWQmgU) I think this is correct.

I am swaying to do a beta release as it is a pain testing various events under certain conditions. If I do... it'll be released on discord and I'll close this thread along with notable changes.

**Comment by ArranzCNL** [comment]
Beta 1 is available via discord.

**Comment by ChrisPC991** [comment]
> [Discord Link](https://discord.com/invite/TqDzcWQmgU) I think this is correct.
> 
> I am swaying to do a beta release as it is a pain testing various events under certain conditions. If I do... it'll be released on discord and I'll close this thread along with notable changes.

Sorry to be a bother but I tried joining the discord link and the discord link says invalid. So could you open a new discord link and could I also have an ETA on release for ImprovedCameraSE which will be compatible with 1.6.6.4.0 Thank you.

**Comment by ArranzCNL** [comment]
@ChrisPC991 discord link is working fine just tested it in my browser. As for official release date cannot give an eta, currently fixing bugs users are finding or that I already know with base game initially. Currently on beta3, should have beta4 out late this evening.
